activerecord/lib/active_record/coders/yaml_column.rb

```diff
@@ -45,14 +45,24 @@ def check_arity_of_constructor
           raise ArgumentError, "Cannot serialize #{object_class}. Classes passed to `serialize` must have a 0 argument constructor."
         end
 
-        def yaml_load(payload)
-          if !ActiveRecord.use_yaml_unsafe_load
-            YAML.safe_load(payload, permitted_classes: ActiveRecord.yaml_column_permitted_classes, aliases: true)
-          else
-            if YAML.respond_to?(:unsafe_load)
+        def safe_load(payload)
+          YAML.safe_load(payload, permitted_classes: ActiveRecord.yaml_column_permitted_classes, aliases: true)
+        end
+
+        if YAML.respond_to?(:unsafe_load)
+          def yaml_load(payload)
+            if ActiveRecord.use_yaml_unsafe_load
               YAML.unsafe_load(payload)
             else
+              safe_load(payload)
```

Comment:
That extra method call will likely eat half of what you gained saving the respond_to, I know the duplication is annoying but I'd just inline it.

Reply:
I did remove the `!` and swapped the if/else while I was at it. It seemed weird to say `if !` with an else. I'm pretty sure there's even a Rubocop rule discouraging that.